### DISCUSSION TITLE
"All" future satelittes

### DISCUSSION THREAD

**Discussion by Martiosk** [Stellarium/stellarium#1639] [Q&A]
Hello! I am a journalist trying to simulate the future night sky over Norway. Is there a plugin with _all_ planned satellites? I found this one, Starlink only, and tried to add it in the source list in the satellite plugin, but the satellites from that link wont show. I am a noob when it comes to these things, so if anyone could, help, I would be grateful! 

http://www.howmanystarlinkswillfillyoursky.com/tle_result 

Best, Martin, Norway

**Comment by gzotti** [comment]
There is the satellites plugin. What you need are orbital elements in the right format. 

Starlink and the other "constellations" increase total number of objects put into orbit in the last 60 years I think about six times within the next very few years. The consequences for scientific astronomy are manifold. None sound desirable.

**Reply by Martiosk** [reply]
Thanks! But i tried to add the tle link from the page mentioned, in the satellite plugin, but without any success. Do you know how to do it?

**Reply by gzotti** [reply]
No, the page returns an internal server error.

**Comment by axd1967** [comment]
Stellarium provides a default Starlink TLE 
https://github.com/Stellarium/stellarium/blob/1e617e49918ab2e5e5fad318dd1ab4024eb1e41c/plugins/Satellites/src/Satellites.cpp#L625

Here is an additional TLE : https://celestrak.com/NORAD/elements/starlink.txt

(I'm not sure what the difference is between the two, but they seem complementary).

You can find more TLE data here: https://celestrak.com/NORAD/elements/

**Reply by Martiosk** [reply]
Thanks! But does this dataset include all the 12.000-ish starlinks? it seems like its only about 2500 or so...?

**Reply by axd1967** [reply]
currently there are only a few thousand Starlink sats in orbit.

https://planet4589.org/space/stats/star/starstats.html

**Reply by Martiosk** [reply]
I know - but all 12000 are simulated in the link above...

**Reply by axd1967** [reply]
(that link is unlikely to be usable as TLE source - it looks like a normal HTML page , although I get a 500 for now).

I think there was a mistake in your link, it should be http://howmanystarlinkswillfillyoursky.com/

updates: 

- indeed, the site provides a full TLE set. but it takes a lot of resources to import the data in Stellarium... (standby, I'm busy with it)
- because there is no progress indicator, it is difficult to judge what is happening; but at least one CPU is 100% busy...
- some of the TLE generate errors such as "Satellite has invalid orbit: "Starlink 9131" "9131""
- it is the writing back of the `satellites.json` file that takes a lot of time
- it looks like the writing back of the JSON file takes a HUGE time (@gzotti @alex-w )
![image](https://user-images.githubusercontent.com/3529789/117136911-485c0300-ada9-11eb-8aa2-b79c85a7fac5.png)
- at first sight, the JSON file seems to have been filled, but Stellarium is still concluding the write operation - or maybe has choked somewhere later. (the "More Satellites" dialog cannot be closed for now)

**Comment by Martiosk** [comment]
Is it possible to upload txt-file from my computer, or does it have to be online? I get an "invalid URL"-error when i try to, even if I have turned off the internet update, as mentioned in the help guide...

**Reply by axd1967** [reply]
you can download text files to your computer, and then include the path in the sat plugin dialog:

![image](https://user-images.githubusercontent.com/3529789/117125729-34f56b80-ad9a-11eb-8bbd-fc4bd731b9ee.png)


(this can also be handy if you want to manually add a few TLE.)

remember that "future" satellites cannot be displayed in Stellarium; AFAIK there are no websites that provide TLE of future sats, but I'm always interested.

**Reply by Martiosk** [reply]
Thanks! Ill try this. 

"remember that "future" satellites cannot be displayed in Stellarium; AFAIK there are no websites that provide TLE of future sats, but I'm always interested."

Have u seen this one? https://www.howmanystarlinkswillfillyoursky.com/ 

Best, Martin

**Comment by Martiosk** [comment]
![image](https://user-images.githubusercontent.com/83653323/117129279-c6ff7300-ad9e-11eb-920c-a63fdd93599d.jpg)
![Uploading image.jpg…]()

**Reply by Martiosk** [reply]
I get this problem though

**Reply by Martiosk** [reply]
![image](https://user-images.githubusercontent.com/83653323/117129493-07f78780-ad9f-11eb-8ecd-474dd1c5eac7.jpg)

**Reply by axd1967** [reply]
you should add a `file:///` but I'm not sure how to do this in Windows. and don't forget to save the new line (yes, the UI is not the most user friendly!)

![image](https://user-images.githubusercontent.com/3529789/117131921-66723500-ada2-11eb-9f50-a2b2fd356222.png)

**Reply by Martiosk** [reply]
I managed, added the "file", it worked. Thanks for your help:) But, its gonna be a lot of satellites up there! Is the stellarium totally realistic? What about light pollution from the city etc? The colour of the satelittes?

**Reply by axd1967** [reply]
whether Stellarium is realistic is a difficult question.
light pollution can be simulated
satellites barely show other than whitish colours. the ISS can sometimes show a more "golden" colour.

**Reply by axd1967** [reply]
@Martiosk did you manage to load and display the data? (see https://github.com/Stellarium/stellarium/discussions/1639#discussioncomment-697707)

**Reply by Martiosk** [reply]
I did, thanks! The future night sky above Norway is dotted with starlink satelittes...

**Comment by meNeither** [comment]
Since it could be of general interest: StarLitter currently simulates up to 76,000 (current and future) satellites from an earthbound perspective: https://starlitter.info/?satellites=satcat&virtual-satellites=all&projection=mollweide&highlight-satellites=true&highlight-virtual-satellites=true&time=&time-mode=time-lapse

I have written a TLE generator based on howmanystarlinkswillfillyoursky.com and try to keep it up to date so that all future satellite constellations can be generated without much effort: https://tle-generator.starlitter.info

A compilation of future mega constellations (Starlink Gen2 Conf1, OneWeb, Kuiper, Telesat and Guowang = ~54,000 satellites) can be found in this file: https://starlitter.info/data/tle/simulated/all.txt (~9.1 MB)

**Reply by gzotti** [reply]
Not sure how many satellites can be efficiently computed and displayed in Stellarium. Bah, I wait for the Kessler1 and Kessler2 belts...

**Reply by meNeither** [reply]
That's one way of dealing with computational limits. 😁

**Reply by axd1967** [reply]
I tried the 12K set and Stellarium balked. I think there is an issue there that needs to be dealt with. On my TODO list...

**Reply by gzotti** [reply]
Just implement satellite computations on the GPU...

**Reply by meNeither** [reply]
Do you know a way to calculate SPG4 on GPU? I was looking for a way and found only this: https://www.researchgate.net/publication/223862732_Using_parallel_computing_for_the_display_and_simulation_of_the_space_debris_environment

**Reply by gzotti** [reply]
No, and I won't do it either. But we accept working contributions.

**Reply by Martiosk** [reply]
Nice, I will try this one too! And probably ask some stellarium for dummies-questions afterwards!
